clients/unshaded/src/main/java/tachyon/client/block/TachyonBlockStore.java

```diff
@@ -80,6 +84,25 @@ public BlockInfo getInfo(long blockId) throws IOException {
   }
 
   /**
+   * @return the info of all active block workers
+   */
+  public List<BlockWorkerInfo> getBlockWorkerInfoList() throws IOException {
+    BlockMasterClient masterClient = mContext.acquireMasterClient();
+    List<BlockWorkerInfo> infoList = Lists.newArrayList();
+    try {
+      for (WorkerInfo workerInfo : masterClient.getWorkerInfoList()) {
+        infoList.add(new BlockWorkerInfo(workerInfo.getAddress().getHost(),
+            workerInfo.getCapacityBytes(), workerInfo.getUsedBytes()));
+      }
+      return infoList;
+    } catch (TachyonException e) {
+      throw new IOException(e);
```

Comment:
This is consistent with the rest of the file, but it isn't great to lose exception type information. Do we have a standard for where to throw the checked `TachyonException` and where to wrap it?